src/cosl/coordinated_workers/coordinator.py

```diff
@@ -434,24 +465,24 @@ def _s3_config(self) -> Dict[str, Any]:
         Raises:
             S3NotFoundError: The s3 integration is inactive.
         """
-        s3_data = self.s3_requirer.get_s3_connection_info()
-        s3_config: Dict[str, Any] = {}
-        if not (
-            s3_data
-            and "bucket" in s3_data
-            and "endpoint" in s3_data
-            and "access-key" in s3_data
-            and "secret-key" in s3_data
-        ):
-            raise S3NotFoundError("s3 integration inactive")
-        s3_config["insecure"] = not s3_data["endpoint"].startswith("https://")
-        s3_config["endpoint"] = re.sub(
-            rf"^{urlparse(s3_data['endpoint']).scheme}://", "", s3_data["endpoint"]
-        )
-        s3_config["region"] = s3_data.get("region", "")
-        s3_config["access_key_id"] = s3_data.pop("access-key")
-        s3_config["secret_access_key"] = s3_data.pop("secret-key")
-        s3_config["bucket_name"] = s3_data.pop("bucket")
+        s3_data = self.s3_connection_info
+        s3_config = {
+            "endpoint": re.sub(rf"^{urlparse(s3_data.endpoint).scheme}://", "", s3_data.endpoint),
+            "region": s3_data.region,
+            "access_key_id": s3_data.access_key,
+            "secret_access_key": s3_data.secret_key,
+            "bucket_name": s3_data.bucket,
+            "insecure": not s3_data.tls_ca_chain,
+            # FIXME: s3 gives us a cert chain. tempo's s3 config takes:
```

Comment:
[now that the worker does it](https://github.com/canonical/cos-lib/pull/87/files#diff-2730d74704f2579d8d7f8f29dd820ba64be312efd1aa5053ec464bc02ee12675R564) is this `FIXME`  still needed?